math/smatrix/inc/Math/CholeskyDecomp.h

```diff
@@ -329,10 +369,10 @@ template<class F> class CholeskyDecompGenDim
     * operator()(int i, int j) for access to elements)
     */
    template<class M> CholeskyDecompGenDim(unsigned N, const M& m) :
-      fN(N), fL(new F[(fN * (fN + 1)) / 2]), fOk(false)
+      fN(N), fL(std::make_unique<F[]>((fN * (fN + 1)) / 2)), fOk(false)
```

Comment:
```suggestion
      fN(N), fL((fN * (fN + 1)) / 2)
```
This would simplify with the `std::vector`. Also, the `fOk(false)` is redundant, because with the change that you made to the class member declaration in this PR, the default value for `fOK` is already `false`.

Reply:
Let's leave the std::vector discussion for the comment above. The fOk is a good point, and I'm happy to revise.